geode-core/src/main/java/org/apache/geode/internal/cache/TombstoneService.java

```diff
@@ -450,6 +450,11 @@ boolean testHook_forceExpiredTombstoneGC(int count, long timeout, TimeUnit unit)
 
     @Override
     protected void beforeSleepChecks() {}
+
+    @Override
+    public long getOldestTombstoneTimeDelta() {
+      return tombstones.peek().getVersionTimeStamp() - cacheTime.cacheTimeMillis();
```

Comment:
The peek could return null...
Will it be a good enough to return version timestamp instead of delta...The caller can calculate delta with different time mechanism (using cacheTime or currentTime).

Also, will it be helpful if the tombstone.toString() info is returned...It has all the information associated with tombstone.

Reply:
I figured that it was one extra step that will always be done, but I will change it.